crates/polars-compute/src/float_sum.rs

```diff
@@ -270,6 +289,18 @@ where
     }
 }
 
+pub fn sum_arr_as_f16<T>(arr: &PrimitiveArray<T>) -> pf16
+where
+    T: NativeType + FloatSum<pf16>,
+{
```

Comment:
I don't really trust the numerical accuracy of this, can we call `sum_arr_as_f32` and cast afterwards?